aws/iam_policy_model.go

```diff
@@ -37,6 +38,22 @@ type IAMPolicyStatementCondition struct {
 type IAMPolicyStatementPrincipalSet []IAMPolicyStatementPrincipal
 type IAMPolicyStatementConditionSet []IAMPolicyStatementCondition
 
+func (self *IAMPolicyDoc) DeDupSids() {
```

Comment:
Nitpick: I think this function would be a little easier to grok if it was something like `Merge(newDoc *IAMPolicyDoc)`, then we can just replace a `*Statement` if the `Sid` is non-empty and matches an existing one otherwise `append()` the new statement. Albeit the below is less optimized since its doing multiple traversals. e.g.

```go
var seen bool
for _, newStatement := range newDoc.Statements {
  if len(newStatement.Sid) == 0 {
    self.Statements = append(self.Statements, newStatement)
    continue
  }
  seen = false
  for i, existingStatement := range self.Statements {
    if existingStatement.Sid == newStatement.Sid {
      self.Statements[i] = newStatement
      seen = true
      break
    }
  }
  if !seen {
    self.Statements = append(self.Statements, newStatement)
  }
}
```

Reply:
I agree, I wasn't super happy with that implementation but changing to a more generic Merge function ended up requiring a heavy refactoring of `dataSourceAwsIamPolicyDocumentRead` without a lot of additional benefit as the model doesn't seem to be used beyond there.

Let me know if you want an additional PR to do that refactoring at some point.